### ISSUE TITLE
Can't perform a React state update on an unmounted component

### ISSUE THREAD

**Issue by liebeskind** [FormidableLabs/nuka-carousel#550]
# Bugs and Questions

### Describe Your Environment

- What version of nuka-carousel are you using?
4.5.5
- What version of React are you using?
16.8.6
- What browser are you using?
Chrome
- What machine are you on?
Macbook

### Describe the Problem
Sometimes when navigating away from page with carousel, I receive the following error in console.  It appears that this might be an issue related to goToSlide, but it doesn't happen every time.
```
Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.

**Comment by feather-jflynn** [comment]
I believe this has something to do with not clearing all timeouts on unmount. Specifically, the timeouts set in this `goToSlide` method: https://github.com/FormidableLabs/nuka-carousel/blob/72fe9b46446ccbb3a2a4e226e643e540caa543d7/src/index.js#L628

**Comment by zgrnk** [comment]
I have been having the same problem, and spent a good amount of time hunting for the bug in my code, before realizing its an issue with nuka-carousel...

Most likely is a separate issue? but similar to this one https://github.com/FormidableLabs/nuka-carousel/issues/100 

It's definitely overkill, but for my use case I found this, and used it as a fix
https://stackoverflow.com/a/8860203
When my carousel component unmounts, I just clear ALL of the timouts

hope this saves someone else some time.

**Comment by sarmeyer** [comment]
fixed in #576

**Comment by nopito** [comment]
Still running into the same issue 😅
